lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2934,6 +2934,22 @@ static AutoDiffParameterIndices *computeDifferentiationParameters(
         builder.setParameter(builder.size() - 1);
         break;
       }
+      case ParsedAutoDiffParameter::Kind::Ordered: {
+        auto index = parsedWrtParams[i].getIndex();
+        if (index >= params.getArray().size()) {
```

Comment:
This is inefficient. Instead, you should declare a `numParams` variable next to where `params` is declared, and use `functionType->getNumParams()` to get it.

Reply:
I'm going to take a look right now, but it seems that `params.getArray().size()` does not always equal `unctionType->getNumParams()`

Reply:
That's why some tests are failing now

Reply:
Using `function->getParameters()->size();` seems to return the expected result versus `functionType->getNumParams()` which seems to give the size of 1 for instance methods due to `self`